cmd/skopeo/sync.go

```diff
@@ -240,6 +253,75 @@ func syncSourceHandler(c *cli.Context, globalDestRef types.ImageReference) (toCo
 	return
 }
 
+func syncSourceFileHandler(c *cli.Context) (toCopyList []imagesBySource, err error) {
+	var cfg sourceCfg
+	if err = cfg.Unmarshal(c.String("source-file")); err != nil {
+		return nil, fmt.Errorf("Unmarshaling failed: %s", err)
+	}
+
+	for domain, domainCfg := range cfg {
+		logrus.Infof("Processing all images for domain: %s", domain)
+
+		for imageName, tags := range domainCfg.Images {
+			var sourceReferences []types.ImageReference
+
+			sourceCtx := &types.SystemContext{
+				RegistriesDirPath:  c.GlobalString("registries.d"),
+				ArchitectureChoice: c.GlobalString("override-arch"),
+				OSChoice:           c.GlobalString("override-os"),
+				DockerCertPath:     domainCfg.CertDir,
+				// DEPRECATED: keep this here for backward compatibility, but override
+				// them if per subcommand flags are provided (see below).
+				DockerInsecureSkipTLSVerify:       !c.GlobalBoolT("tls-verify"),
+				AuthFilePath:                      c.String("authfile"),
+				DockerDaemonCertPath:              domainCfg.CertDir,
+				DockerDaemonInsecureSkipTLSVerify: !domainCfg.TlsVerify,
+			}
```

Comment:
I would rather use one of the different functions that are creating a `types.SystemContext` instance and then change the relevant values with the fields from our `domainCfg`. I would do that to be future proof (see for example the `//DEPRECATED` comment you pasted).

Reply:
Right now there's only one function used to build the context (`contextFromGlobaOptions`). 
So I'll need to create another another function to build the context from `registrySyncCfg`.

Reply:
Why don't you create a `sourceCtx` right after line 271 by using the `contextFromGlobalOptions` and then, inside of the inner for loop (line 274+) you just overwrite the values of `sourceCtx` with the relevant fields from `domainCfg`?